KeeTrayTOTP.Tests/Menu/LegacyTrayMenuTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Windows.Forms;
+using FluentAssertions;
+using KeePass.App.Configuration;
+using KeePass.Forms;
+using KeePass.Plugins;
+using KeePass.UI;
+using KeeTrayTOTP.Menu;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace KeeTrayTOTP.Tests.Menu
+{
+    [TestClass]
+    public class LegacyTrayMenuTests
+    {
+        [TestMethod]
+        public void LegacyTrayMenuItemProvider_ShouldReturnNull()
+        {
+            var plugin = CreatePluginHostMock(out var host);
+            var legacyTrayMenuItemProvider = new LegacyTrayMenuItemProvider(plugin, host.Object);
+
+            var sut = legacyTrayMenuItemProvider.ProvideMenuItem();
+
+            sut.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void LegacyTrayMenuItemProvider_ShouldAddItemsDirectlyToMainWindowsTrayContextMenu()
+        {
+            var plugin = CreatePluginHostMock(out var host);
+            host.Object.CustomConfig.SetBool(KeeTrayTOTPExt.setname_bool_LegacyTrayMenuProvider_Enable, true);
+            var oldItemCount = host.Object.MainWindow.TrayContextMenu.Items.Count;
+            
+            plugin.Initialize(host.Object);
+
+            var sut = host.Object.MainWindow.TrayContextMenu.Items.Count;
+
+            sut.Should().Be(oldItemCount + 2);
+        }
+
+        private static KeeTrayTOTPExt CreatePluginHostMock(out Mock<IPluginHost> host)
```

Comment:
No review comment: This setup is becoming more important as we're adding more tests and this now probably deserves a separate class. I will pick this up in a separate PR.